### ISSUE TITLE
Transform paths from one projection to another

### ISSUE THREAD

**Issue by ahocevar** [openlayers/openlayers#3317]
From @bill-chadwick https://github.com/openlayers/ol3/issues/3285#issuecomment-77351785:

> I also want to factor the basic path interpolation routine ol.geom.flat.geodesic.line_ out of geodesicflatgeom.js as it is a generalised method of converting an a-b path from one projection to another (to a supplied accuracy). I think perhaps, namespace 'ol.geom.flat.path', public function name 'transform' and sourcefile name 'pathflatgeom.js'. geodesicflatgeom.js would then use the transform function in the new namespace. What do you think ?
> 
> The routine works well transforming a British Grid (EPSG:27700) graticule to EPSG:3857.
> I plan also to use it to find parallel paths separated by a fixed distance for the EPSG:3857 graticule.
> 
> The path transform might actually sit best in the ol.proj namespace as a function named transformPath, alongside e.g. transfromExtent.

**Comment by ahocevar** [comment]
Can't you just use `ol.geom.LineString#transform()`? We try to avoid exposing flat coordinate operations to the API, and prefer to operate on geometries or GeoJSON coordinates.

**Comment by bill-chadwick** [comment]
I think ol.geom.LineString.transform just transforms line string vertices, it does not for example, reproject a straight line in one projection to its equivalent curve in another projection.

**Comment by ahocevar** [comment]
What interpolate function do currently use with `ol.geom.flat.geodesic.line_`?

**Comment by bill-chadwick** [comment]
See the demos here 

http://www.bdcc.co.uk/ol3/examples/bing-maps.html

and 

http://www.bdcc.co.uk/ol3/examples/os-graticule.html

here which I updated last night.

They include a link to an updated http://www.bdcc.co.uk/ol3/ol/ol/geom/flat/geodesicflatgeom.js

They still have a few issues to iron out (at low zooms).

For the EPSG:3857 graticule I plan to rework it so that the centre meridian and parallel are drawn first and then further graticule lines are at a fixed distance interval from those. That will need a custom interpolate function.

**Comment by ahocevar** [comment]
It would be easier to look at a diff. Can you create a feature branch with your work and push it to GitHub?

The reason why I asked is because the interpolate functions currently used do not just interpolate a number of points on a straight line between the start and end coordinates, and I'm wondering what the use case for an exposed interpolatePath (or whatever) function would be.

**Comment by bill-chadwick** [comment]
I will do a branch later. 

All I have changed for now is to add a pair of meridian and parallel functions that take a source projection parameter as well as a destination projection parameter. The orginal meridian and parallel functions assumed that the source projection was EPSG:4326.

**Comment by bill-chadwick** [comment]
Perhaps you can help me a bit with github - I am a bit of a newbie with it.

I already have a fork at https://github.com/bill-chadwick/ol3 which I want to update to the current master. Can I do that from the github web portal ?

Can a make a new fork from the web portal e.g. https://github.com/bill-chadwick/ol3-graticule ?

I am developing on Windows.

**Comment by ahocevar** [comment]
You should be able to just do
```
git pull --rebase git://github.com/openlayers/ol3.git master
```

**Comment by bill-chadwick** [comment]
I think I have finally managed to make a 'graticule' branch on my 'fork'. 

See here

https://github.com/openlayers/ol3/compare/master...bill-chadwick:graticule

For the first stab at changes I made to src/ol/geom/flat/geodesicflatgeom.js

I use the two new functions to transform EPSG:27700 grid lines to Web Mercator for plotting on a Web Mercator map. The fit of this grid is very good, all across the Bing OS data.

To draw for example, EPSG:3857 vertical grid lines a fixed distance on the ground apart, I would call the newly public ol.geom.flat.geodesic.transform passing a custom function that returns a new point at a fixed horizontal distance from the grid's central meridian. Such grid lines would not be straight or vertical when zoomed well out.
